src/views/studio/MyVideosView/MyVideosView.tsx

```diff
@@ -0,0 +1,180 @@
+  const { currentPage, setCurrentPage } = usePagination(currentTab)
+  const videosPerPage = ROWS_AMOUNT * videosPerRow
+  const currentTabName = TABS[currentTab]
+  const isPublic_eq = getPublicness(currentTabName)
+  const { loading, videos, error, totalCount, fetchMore } = useVideosOffsetLimitPagination({
+    limit: videosPerPage,
+    offset: videosPerPage * currentPage,
+    where: {
+      channelId_eq: testChannelId,
+      isPublic_eq,
+    },
+  })
+  useEffect(() => {
+    if ((typeof totalCount === 'number' && totalCount > 0) || drafts.length > 0) {
+      setHasAnyVideos(true)
+    } else if (typeof totalCount === 'number' && hasAnyVideos === undefined) {
+      setHasAnyVideos(false)
+    }
+  }, [drafts.length, hasAnyVideos, totalCount])
+
+  // hook to tests draft should be deleted before final
+  useEffect(() => {
+    const createDrafts = async () => {
+      await removeAllDrafts()
+      for (let i = 0; i < 32; i++) {
+        await addDraft({
+          channelId: testChannelId,
+          title: i.toString(),
+          description: 'string',
+          isPublic: true,
+          hasMarketing: false,
+          isExplicit: false,
+        })
+      }
+    }
+    createDrafts()
+    return () => {
+      removeAllDrafts()
+    }
+  }, [])
+
+  const isDraftTab = currentTabName === 'Drafts'
+  const isLoading = loading || (videos?.length === 0 && (totalCount ?? 0) > 0)
+  const placeholderItems = Array.from({ length: isLoading ? videosPerPage : 0 }, () => ({
+    id: undefined,
+    progress: undefined,
+  }))
+  const videosWithPlaceholders = [...(videos || []), ...placeholderItems]
+  const handleOnResizeGrid = (sizes: number[]) => setVideosPerRow(sizes.length)
+  const hasAnyVideo = currentTabName === 'All Videos' && totalCount === 0 && drafts.length === 0
+
+  const handleChangePage = (page: number) => {
+    setCurrentPage(page)
+    currentTabName !== 'Drafts' &&
+      fetchMore({
+        variables: {
+          offset: videosPerPage * page,
+          limit: videosPerPage,
+        },
+      })
+  }
+
+  const gridContent = (
+    <>
+      {isDraftTab
+        ? drafts
+            // pagination slice
+            .slice(videosPerPage * currentPage, currentPage * videosPerPage + videosPerPage)
+            .map((draft, idx) => (
+              <VideoPreviewPublisher
+                key={idx + '-' + currentTabName + '-' + currentPage}
+                id={draft.id}
+                showChannel={false}
+                isDraft
+                isPullupDisabled={false}
+                onEditVideoClick={() => ({})}
+                onDeleteVideoClick={() => {
+                  removeDraft(draft.id)
+                }}
+              />
+            ))
+        : videosWithPlaceholders.map((video, idx) => (
+            <VideoPreviewPublisher
+              key={idx + '-' + currentTabName}
+              id={video.id}
+              showChannel={false}
+              isLoading={loading}
```

Comment:
Do we need that prop? I think the `id` for empty items will be undefined, so the component will enter the loading state anyway? I think this prop may not be needed at all